appcues/src/main/java/com/appcues/trait/appcues/SkippableTrait.kt

```diff
@@ -123,37 +160,171 @@ internal class SkippableTrait(
+                ButtonAppearance.MINIMAL ->
                     Modifier
-                        .padding(buttonAppearance.padding)
-                        .clip(CircleShape)
-                        .drawBehind {
-                            xShapePath(
-                                color = Color(color = 0xFFB2B2B2),
-                                pathSize = buttonAppearance.size,
-                                strokeWidth = 1.5.dp,
-                            ).also { drawPath(path = it, color = Color.Transparent, blendMode = BlendMode.Difference) }
+                        .width(buttonWidth.dp)
+                        .height(buttonHeight.dp)
+                        .handleSkipClick()
+                        .drawMinimalX(isDark)
+            }
+        )
+    }
+
+    // default X button
+    // - background shape with a background color from style or a default semi-transparent 0x54000000
+    // - foreground X path with color from style or a default 0xFFEFEFEF
+    private fun Modifier.drawDefaultX(isDark: Boolean): Modifier {
+        return this.then(
+            background(buttonStyle?.backgroundColor?.getColor(isDark) ?: Color(color = 0x54000000))
+                .drawBehind {
+                    xShapePath(pathSize = (buttonHeight * BUTTON_X_SCALE_FACTOR).dp).also {
+                        drawPath(
+                            path = it,
+                            color = buttonStyle?.foregroundColor?.getColor(isDark) ?: Color(color = 0xFFEFEFEF),
+                            style = Stroke(strokeWidth.toPx()),
+                        )
+                    }
+                }
+        )
+    }
+
+    // minimal X button
+    // - no background
+    // - foreground color uses 70% white tint with an exclusion blend by default, but a specific color from style can override
+    // - an optional shadow can be applied to the path of the X itself (not a background container shadow like default)
+    private fun Modifier.drawMinimalX(isDark: Boolean): Modifier {
+        return this.then(
+            drawBehind {
+                xShapePath(pathSize = (buttonHeight * BUTTON_X_SCALE_FACTOR).dp).also {
+                    drawPathShadow(it, isDark)
+                    val foregroundColor = buttonStyle?.foregroundColor?.getColor(isDark)
+                    val drawStyle = Stroke(strokeWidth.toPx())
+                    if (foregroundColor != null) {
+                        drawPath(
+                            path = it,
+                            color = foregroundColor,
+                            style = drawStyle
+                        )
+                    } else {
+                        drawPath(
+                            path = it,
+                            color = Color.Black,
+                            colorFilter = ColorFilter.tint(Color(color = 0xB3FFFFFF), BlendMode.Difference),
+                            style = drawStyle
+                        )
+                    }
```

Comment:
important part related to the `xShapePath` change - now I can draw the shadow first `drawPathShadow`, then draw the actual path, based on the given style properties. Previously, I could not get it to work, since `xShapePath` was actually drawing the path internally.

Reply:
yes, given the customizations required for this new version it makes sense to change the xShapePath for better flexbility